packages/happy-dom/src/xml-http-request/XMLHttpRequest.ts

```diff
@@ -597,7 +607,11 @@ export default class XMLHttpRequest extends XMLHttpRequestEventTarget {
 			};
 			this._state.status = response.statusCode;
 			this._state.statusText = response.statusMessage;
+			// Although it will immediately be set to loading,
+			// According to the spec, the state should be headersRecieved first.
+			this._setState(XMLHttpRequestReadyStateEnum.headersRecieved);
 			// Sync responseType === ''
+			this._setState(XMLHttpRequestReadyStateEnum.loading);
```

Comment:
but in this #789, it is difficult to implement the test unit due to implementation problems that make this part of the change happen "in a flash" and almost negligible.